Remove FD parameters from DM fitting

HendrikGC02 opened the pull request:
<!--
# nf-core/meerpipe pull request

Many thanks for contributing to nf-core/meerpipe!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

Remember that PRs should be made against the dev branch, unless you're preparing a pipeline release.

Learn more about contributing: [CONTRIBUTING.md](https://github.com/nf-core/meerpipe/tree/master/.github/CONTRIBUTING.md)
-->

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/meerpipe/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/meerpipe _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

HendrikGC02 committed:
Added warning for when template has fewer frequency channels than archive

HendrikGC02 committed:
Ignore FD parameters when fitting for DM

emmacarli commented:
Can you remove the profile part from this PR, and we can discuss the profile one elsewhere as I think there should be more than a warning

emmacarli commented:
Can you also please test that the current code works in removing both FD and DM derivatives from a good sample of par files (testing the case where there is none or either too)?

HendrikGC02 committed:
Removed warning

HendrikGC02 committed:
Changed how FD and DM parameters are removed

HendrikGC02 commented:
I have removed the profile part and now only have the change to remove FD parameters when fitting for DM. I have changed how that is done and tested it on a .par file from J1911-1114 that had both FD and DMX parameters and all possible combinations of them, it removed them successfully.

emmacarli commented:
This was also tested on the rest of the PTA par files by Hendrik and works correctly. This will reflect a change in PTA par files and templates that will have no FD parameters on the portal.
